Configurations/15-ios.conf

```diff
@@ -0,0 +1,64 @@
+#### iPhoneOS/iOS
+#
+# It takes recent enough XCode to use following two targets. It shouldn't
+# be a problem by now, but if they don't work, original targets below
+# that depend on manual definition of environment variables should still
+# work...
+#
+my %targets = (
+    "ios-xcrun" => {
+        inherit_from     => [ "darwin-common", asm("armv4_asm") ],
+        # It should be possible to go below iOS 6 and even add -arch armv6,
+        # thus targeting iPhone pre-3GS, but it's assumed to be irrelevant
+        # at this point.
+        CC               => "xcrun -sdk iphoneos cc",
+        cflags           => add("-arch armv7 -mios-version-min=6.0.0 -fno-common"),
+        sys_id           => "iOS",
+        perlasm_scheme   => "ios32",
+        disable          => [ "engine" ],
+    },
+    "ios64-xcrun" => {
+        inherit_from     => [ "darwin-common", asm("aarch64_asm") ],
+        CC               => "xcrun -sdk iphoneos cc",
+        cflags           => add("-arch arm64 -mios-version-min=7.0.0 -fno-common"),
+        sys_id           => "iOS",
+        bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
+        perlasm_scheme   => "ios64",
+        disable          => [ "engine" ],
+    },
+    "iossimulator-xcrun" => {
+        inherit_from     => [ "darwin-common" ],
+        CC               => "xcrun -sdk iphonesimulator cc",
+        sys_id           => "iOS",
+        disable          => [ "engine" ],
+    },
+# It takes three prior-set environment variables to make it work:
+#
+# CROSS_COMPILE=/where/toolchain/is/usr/bin/ [note ending slash]
+# CROSS_TOP=/where/SDKs/are
+# CROSS_SDK=iPhoneOSx.y.sdk
+#
+# Exact paths vary with Xcode releases, but for couple of last ones
+# they would look like this:
+#
+# CROSS_COMPILE=`xcode-select --print-path`/Toolchains/XcodeDefault.xctoolchain/usr/bin/
+# CROSS_TOP=`xcode-select --print-path`/Platforms/iPhoneOS.platform/Developer
+# CROSS_SDK=iPhoneOS.sdk
+#
```

Comment:
You know, this would probably be placed in a NOTES file...

Reply:
If anybody feels like putting together one, I wouldn't mind. My knowledge of the platform is somewhat limited. One can say it's too "utilitarian." I can compile, I know how to "launch" "console" binary of my choice from command line, how to pass data like evptests.txt, but I can't actually explain how does it work behind the curtains...

Reply:
Formally speaking ios should be in "50 category", because even though one can compile it, and even execute "console" binaries (don't ask), I don't think that it's actually as usable as it could be. Well, things like EVP are perfectly usable, but trouble is that ios application lives in a sandbox and OpenSSL never looks for files there. In other words things that don't refer to files are usable, while those that do - not so much. It would really be appropriate and useful to hear from seasoned ios application developer[s] about what can be improved and write some wise words to other developers about how to use it.